### PR TITLE
Take two of preliminary packagification steps

### PR DESCRIPTION
**Pull request by javh**
Initial packagification steps, take two.  Changes are largely the same things done in pull request #17, with some modifications (like putting all scripts into `inst/exec` instead of language specific subfolders).

The examples in the quickstart seem to work fine, with appropriate modifications to the file paths.

Install via:
```
remotes::install_github("matsengrp/sumrep@packagify-again")
```

However, this won't build the man pages.  Do a full install with roxygenization (man pages) as follows from the root of the cloned repo:
```
library(devtools)
install_deps(dep=T)
document()
build()
install()
```

Or just click _Build | Install & Restart_ in RStudio.

There are import warnings upon build, but you can ignore them for now. I'll fix them after we merge this in. Next steps are:

1. Update documentation.
2. Fix import warnings.
3. Generate a reference manual docs site.
4. Figure out what does/doesn't need to exported. I just exported everything with a roxygen block for now. I suspect that's overkill.
5. Ensure compatibility with lasts AIRR format compatible versions of alakazam and shazam.

I'll work on the above after we merge this in.

### CODE REVIEW

**Comment by BrandenOlson** [comment]
So I keep getting errors similar to the following:

Error: Dependency package(s) 'pegas','plotmo','viridis' not available.

Not sure if this has to do with any of your changes, but I only get them on this branch and not on `master`.

**Comment by javh** [comment]
Try this:

```
library(devtools)
install_deps(dep=T)
```

**Comment by javh** [comment]
Some notes from working through the import conflicts:

* Where is the function `getCDR3s`?
* Consider replacing `multiplot` with `cowplot::plot_grid`, which is more powerful and would let you drop the `grid` dependency.
* Consider replacing `seqinr` functions with `Biostrings` equivalents (if practical).
* Dependencies `pegas`, `RecordLinkage`, and `viridis` don't appear to be used? At least that I can tell.

**Comment by javh** [comment]
Finished the following:

- Updated `docs/usage.md` with new file loading commands.
- Fixed all the import warnings and dependency issues.
- Checked the alakazam and shazam calls. They seem fine.
- Added install of the sumrep package to the Dockerfile. (Not tested yet.)

I think that's enough for now. Test and merge maybe?

**Comment by BrandenOlson** [comment]
Comments:

(1) It looks like `devtools` needs to be loaded for the following command:
``` remotes::install_github("matsengrp/sumrep@packagify-again") ```
Just a heads-up so we remember to make this clear on the installation page.
(2) I'm getting the following error when running the above command:
``` Error: Failed to install 'sumrep' from GitHub: 
  (converted from warning) package ‘Biostrings’ is not available (for R version 3.6.0) ```

**Comment by javh** [comment]
> package ‘Biostrings’ is not available (for R version 3.6.0)

This is because Biostrings is in BioConductor, rather than CRAN proper. I pushed a change that should hopefully pull dependencies from bioconductor as well.

Though, if Biostrings is the only BioC dependency, then the ideal solution is to remove it and replace the functionality with something from another package. Is Biostrings used for anything other than the `getMotifCount` function?

**Comment by BrandenOlson** [comment]
Ok, I'm getting a successful build now.

> Is Biostrings used for anything other than the getMotifCount function?

No, I'm pretty sure it's only used for `DNAStringSet` and `vcountPattern` in `getMotifCount`.

I'm getting some failed tests in `test.getColdspotCountDistribution` and `test.getHotspotCountDistribution` when running `devtools::test()`, so I'll figure those out and then run the examples in the quickstart. Did you want me to test the Docker build via the updated Dockerfile?

**Comment by javh** [comment]
Okay, I'll look at `Biostrings::vcountPattern` and see if it's the same thing as `alakazam::countPatterns`.

I didn't check the tests. Oops.

And, yes on the docker build, but let me build first to make sure the install command I added worked. I'll do that right now and update as soon as it finishes.

**Comment by BrandenOlson** [comment]
The hot and cold spot count tests required a change in the logic (i.e. I don't think your changes broke them) but I've pushed a fix.

The partis tests seem to be breaking because the `test_data.fa` file was removed.

**Comment by javh** [comment]
> The partis tests seem to be breaking because the test_data.fa file was removed.

This is embedded in the package now under `inst/extdata`, the path to which can be retrieved in R via:

```
system.file("extdata/test_data.fa", package="sumrep")
```

If this needs to be accessible outside R (or a wrapper called from R), then maybe it should be copied into `Examples`?

**Comment by javh** [comment]
BTW - Docker image isn't building for me. Think I've gotten past the firewall download issues and am getting actual installation errors now... Still working on it.

**Comment by BrandenOlson** [comment]
Ok, I figured out the partis test issues and it looks fine to me. I'll wait to hear back about the Docker testing and we should be good to merge.

**Comment by javh** [comment]
I think I have all the docker issues fixed, but I need to test a complete start-to-finish build. That'll take some time, so I'll update later.

**Comment by javh** [comment]
Docker build now works for me.

File: Dockerfile [hunk 1]
```diff
@@ -95,21 +95,23 @@ WORKDIR /partis
...
-RUN R --vanilla --slave -e \
-  'install.packages(c("alakazam", "ape", "BiocManager", "CollessLike", "data.table", "dplyr", "entropy", "HDMD", "jsonlite", "magrittr", "Peptides", "RecordLinkage", "shazam", "seqinr", "stringdist", "stringr", "testthat", "textmineR", "yaml"), repos = "http://cran.us.r-project.org")' && \
-  R --vanilla --slave -e 'BiocManager::install("Biostrings")'
+RUN Rscript -e 'install.packages(c("TreeSim", "TreeSimGM"), repos="http://cran.rstudio.com")'
 
-WORKDIR ..
-RUN git clone https://github.com/matsengrp/sumrep.git
-COPY . /sumrep
+# Download sumrep
+RUN git clone --single-branch --branch packagify-again https://github.com/matsengrp/sumrep.git /sumrep
```
Comment (BrandenOlson) [inline Dockerfile]: Will need to update this when merged into master

**Comment by BrandenOlson** [comment]
Okay @javh , I was able to build the Docker image and run a test script successfully (with a small tweak to the dockerfile).

This looks good to me. You have my 👍  to merge whenever you're ready. Lmk if I can help.

**Comment by javh** [comment]
Great! I'll merge sometime before Thanksgiving and switch the branch in the docker file when I do.

**Comment by javh** [comment]
Merged in. Running a test on the docker build from master and I'll push the changes once that passes.

Now that we've merged, I'll try to come back to the documentation tasks in a couple weeks (vignettes, man pages). I'll pester you then about what functions (don't) need to be exported and such.

**Pull request merged**